### PR TITLE
Workflow fail if black and isort fail

### PR DESCRIPTION
**Pull request by UmairK5669**
**Description:**
Ensure that the workflow fails when black and isort style guides are not met. 

**Related Issues:**
#36 

**Tips for the Reviewer:**
Push black and isort breaking changes to observe workflow. 

Note: I was unsure if black and isort should still run as they did before, kept them as is for now however, that can be removed as needed. 

**Checklist:**
Please be sure to check all boxes honestly. This is to ensure a smooth development process, and to reduce the 
likelihood of needing to make additional changes later on.
- [X] I understand that changes authored by individuals who have not signed a Contributor License Agreement (CLA), 
or whose authorship we cannot verify, may not be accepted.
- [X] These contribution address a triaged issue.
- [X] I have performed a self-review of my code and my changes generate no new warnings.
- [X] I have included unit tests for my code contributions, and all tests are passing.
- [X] The docstrings are complete, include doctests demonstrating usage, and are properly formatted. 
I have verified that any updates to the project documentation are complete and look okay.
- [X] I have confirmed all my code contributions are formatted in accordance with the project's Flake8, Black, 
and isort style configurations.
- [X] No unnecessary changes have been made to the Poetry lock file, but `pyproject.toml` and the Poetry lock file 
have been updated to reflect any dependency changes.
- [X] I acknowledge that upon the submission of this pull request, Qoherent will assume the copyright for this 
contribution.

### CODE REVIEW

**Comment by UmairK5669** [comment]
Seems that when you use the `use_pyproject: true` option in the workflow, it fails as it can't find `black` as a dependency in `pyproject.toml`. Check out the [workflow summary](https://github.com/qoherent/ria/actions/runs/8759510074) of [this](https://github.com/qoherent/ria/pull/38/commits/41947d237def027b2f1f3b08696cba3c449820d6) commit to see more. 

Without this flag, it seems our configuration outlined in `pyproject.toml` is still used (ignore files, max-line length etc) and runs without issues. Check out [this](https://github.com/qoherent/ria/actions/runs/8759748795/job/24043400265?pr=38) workflow summary and expand the "Check Black formatting", which shows Black checking for formatting issues with our configuration. Screenshot for reference provided below: 

Screenshot 2024-04-19 at 5 33 59 PM

File: .github/workflows/format.yml [hunk 1]
```diff
@@ -33,9 +33,19 @@ jobs:
         - name: Run Black
           run: black ./ --config pyproject.toml
```
Comment (mrl280) [inline .github/workflows/format.yml:34]: I don't think this section is required. The concern is if we format the code before checking it, then the workflow will always pass.

File: .github/workflows/format.yml [hunk 2]
```diff
@@ -33,9 +33,19 @@ jobs:
...
 
+        - name: Check Black formatting
+          uses: psf/black@stable
+          with:
+            options: "--check --verbose"
+            src: "./"
+            # use_pyproject: true, should be used but causing issues so commenting for now
+
         - name: Run Isort
           run: isort .
```
Comment (mrl280) [inline .github/workflows/format.yml:44]: Same comment as above, I don't think we want to keep this section any more.

File: .github/workflows/format.yml [hunk 3]
```diff
@@ -33,9 +33,19 @@ jobs:
         - name: Run Black
           run: black ./ --config pyproject.toml
 
+        - name: Check Black formatting
+          uses: psf/black@stable
+          with:
+            options: "--check --verbose"
+            src: "./"
+            # use_pyproject: true, should be used but causing issues so commenting for now
```
Comment (mrl280) [inline .github/workflows/format.yml:41]: Can you provide more details as to what's up here? We don't have to resolve the issue in this PR, but any comments might help the next person. Perhaps we should make an issue for this?

**Review by mrl280 — CHANGES_REQUESTED** [review]
Great progress! Perhaps once we remove the "Run Black" and "Run Isort" pieces, we can also remove some of the installs? Not sure if they are required for the "check formatting" sections?

**Comment by mrl280** [comment]
> Seems that when you use the `use_pyproject: true` option in the workflow, it fails as it can't find `black` as a dependency in `pyproject.toml`. Check out the [workflow summary](https://github.com/qoherent/ria/actions/runs/8759510074) of [this](https://github.com/qoherent/ria/pull/38/commits/41947d237def027b2f1f3b08696cba3c449820d6) commit to see more.
> 
> Without this flag, it seems our configuration outlined in `pyproject.toml` is still used (ignore files, max-line length etc) and runs without issues. Check out [this](https://github.com/qoherent/ria/actions/runs/8759748795/job/24043400265?pr=38) workflow summary and expand the "Check Black formatting", which shows Black checking for formatting issues with our configuration. Screenshot for reference provided below:
> 

It looks like the `use_pyproject` option might be about the version of Black to use, rather than the Black configuration to use.

It looks like the version of Black the action will use can be read from the `pyproject.toml` file, and if we set `use_pyproject: true`, it will look into the `tool.black.required-version` field in the `pyproject.toml` file, which we don't have.

As it is now, do we know whether its using the Black configuration in `pyproject.toml`? It might be using it by default?

**Comment by UmairK5669** [comment]
> As it is now, do we know whether the workflow is using the Black configuration in `pyproject.toml`? With any luck it might be using it by default?

I think we got lucky haha as black's default config is set to 88 characters per line max, and as we can see from the workflow summary the max line length being used is 119 characters. Check it out yourself [here](https://github.com/qoherent/ria/actions/runs/8761555646/job/24048213027?pr=38) (expand the 'Check Black formatting' section.

Screenshot 2024-04-19 at 9 27 33 PM

Also, I removed the force format using black and isort, so the workflow will fail if the code doesn't meet those standards just like with flake8. The `pip install black` and `pip install isort` were also removed. 

I believe this should now be good to go.

**Review by mrl280 — APPROVED** [review]
Looks good✔️

**Pull request merged**